File: go/day0/solution.go
package day0

import (
	utils "advent-of-code/utils"
	"fmt"
	"os"
	// "math"
	// "os"
	// "strings"
)


func Solve(runAs string) {
	dir, err := os.Getwd()
	utils.CheckError(err)

	var inputFile = fmt.Sprintf("%s/day0/input_%s.txt", dir, runAs)

	println(inputFile)

	// readFile, err := os.Open(inputFile)
    // utils.CheckError(err)
    // fileScanner := bufio.NewScanner(readFile)
    // fileScanner.Split(bufio.ScanLines)
  
    // SCORE_I := 0
    // SCORE_II := 0
	
    // for fileScanner.Scan() {
    //     // do something
	// }

	// println("The solution for part I is:", SCORE_I)
	// println("The solution for part II is:", SCORE_II)
}

File: rust/src/day0/solution.rs
use std::{
    fs::File,
    io::{self, BufReader, Lines},
};

use crate::utils::read_lines_iterable;

pub fn solve(run_as: char) -> Result<(), io::Error> {
    let input_file = format!("src/day0/input_{run_as}.txt");

    let lines = read_lines_iterable(input_file.clone())?;

    let mut count = part1(lines);
    println!("Part 1: {}", count);

    let lines2 = read_lines_iterable(input_file)?;
    count = part2(lines2);
    println!("Part 2: {}", count);

    Ok(())
}

pub fn part1(lines: Lines<BufReader<File>>) -> u32 {}

pub fn part2(lines: Lines<BufReader<File>>) -> u32 {}
